### PR TITLE
fix Issue 20716 - Wrong code/ABI for extern(C++) interface method tha…

### PR DESCRIPTION
**Pull request by WalterBright**
…t returns non-POD

### CODE REVIEW

**Comment by WalterBright** [comment]
This is ready to go.

**Comment by UplinkCoder** [comment]
The commit message is too short.
Just reading the changes don't tell me what's going on.

**Comment by UplinkCoder** [comment]
@WalterBright This should be part of the commit message: 
The problem here is the "adjustor thunk" generated does not take into account the size of the hidden pointer parameter, and winds up adjusting the hidden pointer rather than the `this` pointer.

**Comment by thewilsonator** [comment]
Stable.

File: src/dmd/backend/cod3.d [hunk 1]
```diff
@@ -4719,6 +4719,8 @@ void cod3_thunk(Symbol *sthunk,Symbol *sfunc,uint p,tym_t thisty,
         p += I32 ? 8 : tysize(TYfptr);          // far function
     else
         p += tysize(TYnptr);
+    if (tybasic(sfunc.ty()) == TYhfunc)
+        p += tysize(TYnptr);                    // skip over hidden pointer
```
Comment (RazvanN7) [inline src/dmd/backend/cod3.d:4723]: This line appears to not be covered by tests.

**Review by RazvanN7 — APPROVED** [review]
Rebase to stable and add a test to cover the uncovered line and we are good to go.

**Pull request merged**